### PR TITLE
feat: Added interface for swagger options

### PR DESCRIPTION
**Pull request by seebeen**
Options object defined from this page: https://swagger.io/docs/open-source-tools/swagger-ui/usage/configuration/

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Other... Please describe: 

## What is the current behavior?
Currently, Swagger Options have no interface object, meaning that anyone that wants to customize swagger needs to google the options, and read stackoverflow answers in order to learn how to customize
Issue Number: N/A

## What is the new behavior?
I've added a documented interface from swagger docs with some commonly used options

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: lib/interfaces/swagger-custom-options.interface.ts [hunk 1]
```diff
@@ -2,9 +2,111 @@ interface CommonSwaggerCustomOptions {
   useGlobalPrefix?: boolean,
 }
 
+export interface SwaggerConfigurationOptions {
```
Comment (micalevisk) [inline lib/interfaces/swagger-custom-options.interface.ts:5]:
```suggestion
/**
 * Check out all available `swagger-ui` options here:
 * {@link https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md}
 */
export interface SwaggerConfigurationOptions {
```

**Review by micalevisk — CHANGES_REQUESTED** [review]
well done!

can you do `npm run format` (and commit just `swagger-custom-options.interface.ts`)

**Comment by jmcdo29** [comment]
Are all of these options available for `swagger-ui-express` and `fastify-swagger`?

**Comment by seebeen** [comment]
> Are all of these options available for `swagger-ui-express` and `fastify-swagger`?

As far as I'm aware these are native swagger options you can use for both express / fastify because they're mostly used for handling the UI side of things.

I've tested some of the options (persistAuth, docExpansion) on fastify. Works as expected.

**Comment by seebeen** [comment]
> well done!
> 
> can you do `npm run format` (and commit just `swagger-custom-options.interface.ts`)

Done :)

**Comment by seebeen** [comment]
@jmcdo29 All good? Can the PR be merged? :)

**Comment by kamilmysliwiec** [comment]
This PR does not seem to be relevant anymore after merging this PR https://github.com/nestjs/swagger/pull/1886

**Pull request closed without merging**